File: solutions/0019-remove-nth-node-from-end-of-list/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode temp=head,temp1=head;
        while(n-- >0)
            temp=temp.next;
        if(temp==null)
            return head.next;
        while(temp.next!=null)
        {
            temp=temp.next;
            temp1=temp1.next;
        }
        if(temp1.next!=null)
            temp1.next=temp1.next.next;
        return head;
    }
}

File: solutions/0019-remove-nth-node-from-end-of-list/solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: Optional[ListNode]
        :type n: int
        :rtype: Optional[ListNode]
        """
        temp=head;
        temp1=head;
        while(n>0):
            n-=1
            temp=temp.next;
        if(not temp):
            return head.next
        while(temp.next):
            temp=temp.next
            temp1=temp1.next
        if(temp1.next):
            temp1.next=temp1.next.next
        return head;
